controller/src/command.cpp

```diff
@@ -0,0 +1,371 @@
+
+            watchdog_reboot();
+
+            break;
+
+        /* Mixed mode commands */
+
+        case command::set_periodic:
+
+            parameters_setPeriodicMode((enum periodicMode) dataTx[0]);
+
+            break;
+
+        case command::get_periodic:
+            *lenRx = 1;
+            dataRx[0] = (char) parameters_getPeriodicMode();
+
+            break;
+
+        case command::set_mode:
+
+            parameters_setOperatingMode((enum operatingMode) dataTx[0]);
+
+            break;
+
+        case command::get_mode:
+
+            *lenRx = 1;
+            dataTx[0] = (char) parameters_getOperatingMode();
+
+            break;
+
+        case command::comms_check:
+
+            break;
+
+        case command::set_ventilatorMode:
+
+            parameters_setVentilatorMode((enum ventilatorMode) dataTx[0]);
+
+            break;
+
+        case command::get_ventilatorMode:
+
+            *lenRx = 1;
+            dataTx[0] = (char) parameters_getVentilatorMode();
+
+            break;
+
+        case command::start_ventilator:
+
+            break;
+
+        case command::stop_ventilator:
+
+            break;
+
+        case command::set_solenoidNormalState:
+
+            parameters_setSolenoidNormalState((enum solenoidNormaleState) dataTx[0]);
+
+            break;
+
+        default:
+
+            break;
+    }
+}
+
+void command_responseSend(uint8_t cmd, char *packet, uint8_t len) {
+    serialIO_send(msgType::rAck, (enum dataID) cmd, packet, len);
+}
+
+/****************************************************************************************
+*    PRIVATE FUNCTIONS
+****************************************************************************************/
+
+/****************************************************************************************
+*  @brief
+*  @usage
+*  @param
+*  @param
+*  @return
+****************************************************************************************/
+static float convIntTofloat(char *data) {
+
+    float f;
+
+#ifdef LITTLE_ENDIAN
+    /* Convert big endian integer to little endian float */
+
+    uint32_t data_bEndian = 0;
+    uint32_t data_lEndian = 0;
+
+    memcpy(&data_bEndian, data, sizeof(data_bEndian));
+
+    data_lEndian =  __builtin_bswap32(data_bEndian);
+
+    memcpy(&f, &data_lEndian, sizeof(f));
+#else
+    /* Convert big endian to big endian float */
+
+    memcpy(&f, &data, sizeof(f));
+#endif
+
+    return f;
+}
+
+/****************************************************************************************
+*  @brief
+*  @usage
+*  @param
+*  @param
+*  @return
+****************************************************************************************/
+static uint32_t convfloatToInt(float float_value) {
+
+    #ifdef LITTLE_ENDIAN
+        /* Convert float into big endian format for transmission */
+        uint32_t data_bEndian = 0;
+        uint32_t data_lEndian = 0;
+
+        memcpy((void * ) data_lEndian, (void *) &float_value, sizeof(data_lEndian));
```

Comment:
We don't have to cast to void* above; is it necessary here?

Reply:
Lee writes in an email:

> I didn't initially have the casts, but then the compiler started complaining so I added them.

I see now, it's a bug.  It should be `&data_lEndian`, then it will compile without cast to void*.